common/src/main/java/com/firebase/ui/common/Adapter.java

```diff
@@ -0,0 +1,35 @@
+package com.firebase.ui.common;
+
+import android.arch.lifecycle.LifecycleObserver;
+import android.support.annotation.NonNull;
+import android.support.annotation.RestrictTo;
+
+@RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)
+public interface Adapter<O extends BaseObservableSnapshotArray<S, E, ?, T>, S, E, T>
+        extends BaseChangeEventListener<S, E>, LifecycleObserver {
```

Comment:
We completely missed documentation for the `FirestoreRecyclerAdapter` so I feel pretty strongly towards having a base `Adapter` interface that we can inherit from to get shared documentation even with those ugly generics. @samtstern What do you think?

Reply:
Just looking at this class ... I can't imagine it meaning anything to anyone but the two of us.  Let's just solve the firestore docs issue by adding those docs directly, this abstraction doesn't seem necessary since we don't take `Adapter` as an argument to anything.

If you feel strongly about this it should be a separate PR anyway, this should stay focused on the `Null`/`NonNull` stuff for Kotlin.

Reply:
@samtstern I do prefer clarity over minimizing copypastas... since we both agree it's ugly and you think other people won't know what's going on, I'm happy to just copypasta. 😄